### PR TITLE
Update "to" prop

### PR DESCRIPTION
**Pull request by Domino987**
To accepts to an object with pathname, search and has values.
https://github.com/ReactTraining/react-router/blob/dev/docs/advanced-guides/migrating-5-to-6.md#note-on-link-to-values

### CODE REVIEW

File: types/react-router/index.d.ts [hunk 1]
```diff
@@ -153,8 +153,17 @@ export const __RouterContext: React.Context<RouteComponentProps>;
 
 export function useLocation<S = H.LocationState>(): H.Location<S>;
 
+export type To =
```
Comment (murbanowicz) [inline types/react-router/index.d.ts]: I would rename it to something more verbose, however I don't have an idea at this moment. What you think?

**Review by murbanowicz — CHANGES_REQUESTED** [review]

File: types/react-router/index.d.ts (hunk 1 above)
Comment (Domino987) [reply types/react-router/index.d.ts]: How about DestinationProp?

Comment (murbanowicz) [reply types/react-router/index.d.ts]: Can be just Desitnation or DestinationProp as you proposed. I will accept any of these two

Comment (Domino987) [reply types/react-router/index.d.ts]: Done

**Review by murbanowicz — APPROVED** [review]
Thank you!

**Pull request merged**